### PR TITLE
feat: dump request headers in verbose mode

### PR DESCRIPTION
**Pull request by alexander-fenster**
Not sure if we have any test coverage for logging, but if we do, please let me know!

### CODE REVIEW

**Comment by alexander-fenster** [comment]
Kotlin CI seems to be failing because of the GCP project being suspended.

**Comment by noahdietz** [comment]
> Kotlin CI seems to be failing because of the GCP project being suspended.

Yep. If you want to change the image in the CI config, feel free, else I will do it separately. 

`gcr.io/gapic-images/gapic-generator-kotlin:0.6.0`

**Comment by noahdietz** [comment]
> Kotlin CI seems to be failing because of the GCP project being suspended.

Yep. If you want to change the image in the CI config, feel free, else I will do it separately. 

`gcr.io/gapic-images/gapic-generator-kotlin:0.6.0`

File: cmd/gapic-showcase/loggers.go [hunk 1]
```diff
@@ -33,13 +34,31 @@ type loggerObserver struct{}
 
 func (l *loggerObserver) GetName() string { return "loggerObserver" }
 
+func DumpIncomingHeaders(ctx context.Context) {
```
[the remark this answers is not in the document]
Comment (alexander-fenster) [reply cmd/gapic-showcase/loggers.go]: Pure magic! You surely could not expect me to know that :)  Fixed, thank you!

**Pull request merged**